### DISCUSSION TITLE
How to 'combine' data when joining tables

### DISCUSSION THREAD

**Discussion by jan-nick** [supabase/supabase#6874] [Questions]
I have the following tables:

**items**

| id         | name |
|--------------|-----------|

**user_items**

| id (pk and fk referring to items)        | user_id|
|--------------|-----------|

Now I want query all user items:

```
const query = supabase
  .from('items')
  .select(`
    *,
    user_items:id(*)
  `)
```

This is what I expected for the returned data:

```
[
  {
    id: 'some-item-id',
    user_id: 'some-user-id',
    name: 'some-item-name',
  },
  ...
]
```

This is what is actually returned:

```
[
  {
    id: 'some-item-id',
    name: 'some-item-name',
    user_items: {
      id: 'some-item-id',
      user_id: 'some-user-id',
    },
  },
  ...
]
```

**Is it possible to modify the select statement, so the columns of the two tables are combined into a single table?**

Especially when using filters this could be helpful. E.g. when filtering for user_id on user-items I would get this error with the current implementation:
```
query.
  .eq('user_items.user_id', 'some-user-id')
```

> `column files.userId does not exist`

**Comment by steve-chavez** [comment]
Right now you'd have to do the transformation client side but we have a [planned enhancement](https://github.com/PostgREST/postgrest/issues/1233#issuecomment-1115118458) that would let you do the following to get your desired result.
```js
const query = supabase
  .from('items')
  .select(`
    *,
    ...user_items(name)
  `)
```

Btw, note that you can do the join by just specifying the table name(`id` can be omitted):

```js
const query = supabase
  .from('items')
  .select(`
    *,
    user_items(*)
  `)
```

**Reply by jopfre** [reply]
I can see that the pr for this has been merged in postgrest. Any idea when the functionality might come to the api?

**Reply by steve-chavez** [reply]
@jopfre This will be announced on the upcoming launch week(next week), then it will be deployed to all projects.

**Reply by steve-chavez** [reply]
This is available locally with the CLI now, see https://supabase.com/blog/postgrest-11-prerelease#spreading-related-tables. For production you can request it to support@supabase.com.

**Reply by lcmchris** [reply]
Hi @steve-chavez - Is this available now in production?

**Reply by steve-chavez** [reply]
@lcmchris Not yet, please subscribe to https://github.com/supabase/postgres/pull/628. I'll announce there when it's available.

**Comment by fabianwaller** [comment]
When doing this 
```
const { data, error } = await supabase
    .from('members')
    .select(`...teams(*)`)
    .eq('user_id', user.id)
```
typescript reports `const data: ParserError<"Expected identifier at `...teams(*)`">[] | null`.
Why it does not detect the correct type?

**Reply by BenHerbst** [reply]
It isn't yet a feature I think, he just said it's going to be there maybe in future

**Reply by steve-chavez** [reply]
@BenHerbst spread is already supported https://postgrest.org/en/la/references/api/resource_embedding.html#spread-embedded-resource

**Reply by fluid-design-io** [reply]
Can confirm the spread feature is added to supabase ✅, though typescript still complains about it.

**Reply by RedChops** [reply]
I've been seeing that too, kind of annoying but I can somewhat figure it out

**Reply by steve-chavez** [reply]
This was merged recently https://github.com/supabase/postgrest-js/pull/497 so perhaps only an update to supabase-js is required.
